sdk/communication/communication-common/src/credential/clientArguments.ts

```diff
@@ -27,8 +27,14 @@ const assertValidEndpoint = (host: string): void => {
  *
  * @param credential - The credential being checked.
  */
-export const isKeyCredential = (credential: any): credential is KeyCredential => {
-  return credential && typeof credential.key === "string" && credential.getToken === undefined;
+export const isKeyCredential = (
+  credential?: KeyCredential | PipelineOptions | TokenCredential
+): credential is KeyCredential => {
+  if (credential === undefined) {
+    return false;
+  }
+
+  return credential && (credential as KeyCredential).key !== undefined;
```

Comment:
why did we relax the condition to no longer check the type of `key` and the absence of `getToken`?